pkg/declarative/indexconfig.go

```diff
@@ -0,0 +1,95 @@
+package declarative
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/operator-framework/operator-registry/pkg/image"
+	libregistry "github.com/operator-framework/operator-registry/pkg/lib/registry"
+	registry "github.com/operator-framework/operator-registry/pkg/registry"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ConfigEditor is an implementation of IndexConfig
+type ConfigEditor struct {
+	Logger      *logrus.Entry
+	ImgRegistry image.Registry
+}
+
+// AddToConfig persists the bundle in it's package config
+func (c ConfigEditor) AddToConfig(request AddConfigRequest) error {
+
+	defer func() {
+		if err := c.ImgRegistry.Destroy(); err != nil {
```

Comment:
We should probably destroy the registry in the same scope that it was created in. This way, we remove at least one side effect of running `AddToConfig`.